astroid/interpreter/_import/spec.py

```diff
@@ -285,7 +296,7 @@ def _precache_zipimporters(path=None):
             pic[entry_path] = zipimport.zipimporter(entry_path)
         except zipimport.ZipImportError:
             continue
-    return pic
+    return {key: value for key, value in pic.items() if isinstance(value, zipimport.zipimporter)}
```

Comment:
I'm not sure if that's as intended, but this simply filters out everything from `pic` that is not a `zipimport.zipimporter` class (i.e. the other types of importers). This may be related to the failing test too.

Reply:
@GergelyKalmar it is exactly as intended. The goal is to have a container with only one type of objects. The failing tests (dealing with pypy) are not due to this PR.